django/utils/functional.py

```diff
@@ -101,14 +91,6 @@ def __prepare_class__(cls):
                             continue
                         meth = cls.__promise__(method_name)
                         setattr(cls, method_name, meth)
-            cls._delegate_bytes = bytes in resultclasses
-            cls._delegate_text = str in resultclasses
-            assert not (cls._delegate_bytes and cls._delegate_text), (
-                "Cannot call lazy() with both bytes and text return types.")
-            if cls._delegate_text:
-                cls.__str__ = cls.__text_cast
-            elif cls._delegate_bytes:
-                cls.__bytes__ = cls.__bytes_cast
```

Comment:
So now we have `__proxy__.__str__` defined as `str(self.__cast())`...

...but `__proxy__.__str__` is not defined at all. Maybe it should be `bytes(self.__cast())`.

Reply:
> Did you mean to write `__proxy__.__bytes__` instead of `__proxy__.__str__` in the second sentence?

Yes, sorry.

> If so, the difference between `__str__` and `__bytes__` relates to the code comment inside `def __str__` below; namely, that the `object` type defines `__str__`, but doesn't define `__bytes__`, so the special treatment that `__str__` gets isn't needed for `__bytes__`.

Ok. I see. I also notice that there a bunch of other things defined on `__proxy__` that override those on `object`.

I wonder if there are any others here that ought to be that are not?
```
['__class__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__']
```

Reply:
From that list, I think the following *might* be considered:

- `__delattr__`, `__setattr__`, `__getattribute__`: not sure, might be tricky?
- `__reduce_ex__`: not sure what this is, but we already forward `__reduce__`, so probably should forward this one too?
- `__format__`: should forward this, example:

```py
>>> class CustomFormat:
...    def __format__(self, format_spec):
...       return 'custom'
... 
>>> format(CustomFormat(), 'x')
'custom'
>>> format(lazy(lambda: CustomFormat(), CustomFormat)(), 'x')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
TypeError: unsupported format string passed to __proxy__.__format__
```